tests/voq/test_voq_nbr.py

```diff
@@ -32,6 +37,134 @@
 NEW_MAC = "00:01:94:00:00:01"
 
 
+def check_bgp_restored(duthosts, all_cfg_facts):
+    """
+    Checks for bgp neighbors to be established.
+
+    Args:
+        duthosts: duthosts fixture
+        all_cfg_facts: fixture from voq/conftest.py
+
+    Returns:
+        True if all neighbors are established, False if not.
+    """
+    down_nbrs = 0
+    for node in duthosts.frontend_nodes:
+        for asic in node.asics:
+            asic_cfg_facts = all_cfg_facts[node.hostname][asic.asic_index]['ansible_facts']
+            if 'BGP_NEIGHBOR' not in asic_cfg_facts:
+                continue
+
+            if node.is_multi_asic:
+                bgp_facts = node.bgp_facts(instance_id=asic.asic_index)['ansible_facts']
+            else:
+                bgp_facts = node.bgp_facts()['ansible_facts']
+
+            for address in asic_cfg_facts['BGP_NEIGHBOR'].keys():
+                if bgp_facts['bgp_neighbors'][address]['state'] != "established":
+                    logger.info("BGP internal neighbor: %s is down: %s." % (
+                        address, bgp_facts['bgp_neighbors'][address]['state']))
+                    down_nbrs += 1
+    if down_nbrs != 0:
+        logger.warning("Neighbors are still down: %d", down_nbrs)
+        return False
+    else:
+        logger.info("All BGP neighbors are restored.")
+        return True
+
+
+def restore_bgp(duthosts, nbrhosts, all_cfg_facts):
+    """
+    Attempts to restore BGP config.
+
+    Args:
+        duthosts: duthosts fixture
+        nbrhosts: nbrhosts fixture
+        all_cfg_facts: fixture from voq/conftest.py
+
+    """
+    for node in duthosts.frontend_nodes:
+        for asic in node.asics:
+            asic_cfg_facts = all_cfg_facts[node.hostname][asic.asic_index]['ansible_facts']
+            if 'BGP_NEIGHBOR' not in asic_cfg_facts:
+                continue
+
+            if node.is_multi_asic:
+                bgp_facts = node.bgp_facts(instance_id=asic.asic_index)['ansible_facts']
+            else:
+                bgp_facts = node.bgp_facts()['ansible_facts']
+
+            for address in asic_cfg_facts['BGP_NEIGHBOR'].keys():
+                if bgp_facts['bgp_neighbors'][address]['state'] == "established":
+                    logger.info("BGP internal neighbor: %s is established: %s, no action." % (
+                        address, bgp_facts['bgp_neighbors'][address]['state']))
+                else:
+                    logger.info("BGP internal neighbor: %s is not established: %s, will enable" % (
+                        address, bgp_facts['bgp_neighbors'][address]['state']))
+
+                    logger.info(
+                        "Startup neighbor: {} on host {} asic {}".format(address, node.hostname, asic.asic_index))
+                    asnum = asic_cfg_facts['DEVICE_METADATA']['localhost']['bgp_asn']
+                    node.command("sudo config bgp startup neighbor {}".format(address))
+                    if node.is_multi_asic:
```

Comment:
why we need to do this? the startup neighbour command in line 109 is doing the same thing

Reply:
Fixed in latest.